### PR TITLE
add application component name duplication check in webhook

### LINKED ISSUE #1703: [BUG] cannot parser same resource
### Demo
```
apiVersion: core.oam.dev/v1beta1
kind: ComponentDefinition
metadata:
  name: redis-cluster
  namespace: default
  annotations:
    definition.oam.dev/description: "redis-cluster component"
spec:
  workload:
    definition:
      apiVersion: apps/v1
      kind: StatefulSet
  schematic:
    cue:
      template: |
        parameter: {
        	// +usage=名称
        	name: string
        	// +usage=端口号
        	port: *6379 | int
        	// +usage=密码
        	password: string
        	// +usage=镜像地址
        	image: *"docker.io/bitnami/redis:6.2.3-debian-10-r15" | string
        	// +usage=存储大小
        	storage: *"8Gi" | string
        	// +usage=是否添加nodeservice
        	nodeService: *false | bool
        	// +usage=副本数量
        	replicasCount: *3 | int
        	replicasCount: (>=1 & <=10)
        	// +usage=副本规格
        	replicasResource: "0.5C0.5G" | "1C2G" | *"2C4G" | "4C8G" | "nolimit"
        	// +usage=master 规格
        	masterResource: *"4C8G" | "8C16G" | "nolimit"
        }

        //serviceAccount
        output: {
        	"apiVersion": "v1"
        	"kind":       "ServiceAccount"
        	"metadata": {
        		"name":  "sa-redis-" + parameter.name
        		"labels": {
        			"app.kubernetes.io/name": "sa-redis-" +parameter.name
        		}
        	}
        }

        outputs:test: {
        	"apiVersion": "v1"
        	"kind":       "ServiceAccount"
        	"metadata": {
        		"name":  "sa-redis22-" + parameter.name
        		"labels": {
        			"app.kubernetes.io/name": "sa-redis-" +parameter.name
        		}
        	}
        }
```
### Appfile
```
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: redis-dev
  namespace: default
spec:
  components:
    - name: redis-dev
      type: redis-cluster
      properties:
        password: "123456"
        name: "redis-dev"
        masterResource: "nolimit"
        replicasResource: "nolimit"
        replicasCount: 1
        storage: "500m"
```

### result
```
➜  ~ kubectl get sa
NAME                   SECRETS   AGE
redis-dev              1         2s
sa-redis22-redis-dev   1         2s
```
### want 
i want have two serviceAccount: sa-redis-redis-dev and sa-redis22-redis-dev

### PR DESCRIPTION
**Pull request by Chwetion**
Fixes #1703 
Add the repeat verification of application component names to the webhook to compare with the existing application, but will not compare with the component resources created by no applications.

### CODE REVIEW

**Review by wonderflow — APPROVED** [review]
LGTM

File: pkg/appfile/parser.go [hunk 1]
```diff
@@ -238,6 +238,38 @@ func (p *Parser) parseTrait(ctx context.Context, name string, properties map[str
 	}, nil
 }
 
+// ValidateComponentNames validate all component name whether repeat in cluster and template
+func (p *Parser) ValidateComponentNames(ctx context.Context, af *Appfile) (int, error) {
+	existCompNames := make(map[string]string)
+	existApps := v1beta1.ApplicationList{}
+	if err := p.client.List(ctx, &existApps); err != nil {
+		return 0, err
```
Comment (wonderflow) [inline pkg/appfile/parser.go:246]: ignore not exist？

Comment (Chwetion) [reply pkg/appfile/parser.go:246]: existApps will empty when not exist, this err may connect apiserver error or list permission denied

**Comment by yangsoon** [comment]
LGTM

**Pull request merged**